pants_jupyter_plugin/plugin.py

```diff
@@ -381,8 +386,26 @@ def pants_repo(self, pants_repo: str):
       self._display_line(f'ERROR: could not find a valid pants repo at {pants_repo}\n')
       return
 
-    self._display_line(f'Using pants repo at: {pants_repo}\n')
-    self._pants_repo = pants_repo.absolute()
+    # Version check for pants v1 vs v2 flags/behavior.
+    with temporary_dir() as tmp_dir:
+      stderr_path = pathlib.Path(tmp_dir) / 'stderr'
+      try:
+        with open(stderr_path, 'w') as stderr:
+          version_string = subprocess.check_output(
+            ['./pants', '--version'],
+            stdin=subprocess.DEVNULL,
+            stderr=stderr,
+            cwd=pants_repo,
+          ).decode().strip()
+      except:
+        if stderr_path.is_file():
+          print(f"`pants --version` failed with:\n{stderr_path.read_text()}", file=sys.stderr)
+        raise
+    is_pants_v2 = version_string.startswith("2")
+
+    self._display_line(f'Using pants {version_string} in repo at: {pants_repo}\n')
+    pants_repo = pants_repo.absolute()
+    self._pants_repo = _PantsRepo(pants_repo, is_pants_v2)
```

Comment:
Oops - missed this in 1st review pass. I think the "...repo definition such that there's a one-time cost." bit is still dangling. It would probably be good to include a one-time guard in this change.

Reply:
This method runs explicitly when someone runs `%pants_repo $path`, which is a one time operation.